url.go

```diff
@@ -229,9 +202,7 @@ func ParseURL(url string) (*URL, error) {
 }
 
 func parseV1URL(url *gourl.URL, originalURL string) (*URL, error) {
-	r := URL{
-		Schema: CharmStore.String(),
-	}
+	r := URL{}
 	if url.Scheme != "" {
 		r.Schema = url.Scheme
 	}
```

Comment:
Officially, we can now close the weird hole here. Previously, if you passed in `ftp://` it would put that as the front of the charm URL. I don't think we'll ever want that. Instead, I think we should be a lot stricter and error out when we don't have a local scheme. This was originally how they squeezed in `http` and `https`, which I was never fond of. 

```go

	if url.Scheme != "local" {
		return nil, errors.Errorf("expected local scheme: %q", url.Scheme)
	}
    r := URL{}
```